### DISCUSSION TITLE
Basic Web Security 👮‍♂️

### DISCUSSION THREAD

**Discussion by mazipan** [ngobrolin/web#41] [Ideas]
# Main References
- [Website security from MDN](https://developer.mozilla.org/en-US/docs/Learn/Server-side/First_steps/Website_security)
- [OWASP Cheat Sheet Series](https://cheatsheetseries.owasp.org/index.html)

## Things we can discuss
- [Cross-Site Scripting (XSS)](https://developer.mozilla.org/en-US/docs/Learn/Server-side/First_steps/Website_security#cross-site_scripting_xss)
- [SQL injection](https://developer.mozilla.org/en-US/docs/Learn/Server-side/First_steps/Website_security#sql_injection)
- [Cross-Site Request Forgery (CSRF)](https://developer.mozilla.org/en-US/docs/Learn/Server-side/First_steps/Website_security#cross-site_request_forgery_csrf)
- [Clickjacking](https://developer.mozilla.org/en-US/docs/Glossary/Clickjacking)
- [Apa itu serangan DDoS?](https://www.microsoft.com/id-id/security/business/security-101/what-is-a-ddos-attack)
- [File Inclusion Vulnerabilities](https://brightsec.com/blog/file-inclusion-vulnerabilities/)

## Things you should do
- Update dependencies regularly
- Do not save PII (Personally identifiable information) data if not necessary
- Do not push any secrets to repositories
  - Use Vault or Secret Manager
  - Use .gitignore
- Setup HTTPS
- Adding WAF (Web Application Firewall)
- Choose password-less, 2FA/MFA
- Use secure cookie
- [OWASP Top Ten](https://owasp.org/www-project-top-ten/)
- Sanitize your input
- Automated security testing

## Websites/tools
- https://securityheaders.com/

### Vulnerability management tools
- https://owasp.org/www-community/Vulnerability_Scanning_Tools
- [sonarqube](https://www.sonarsource.com/products/sonarqube/)
- [defectdojo](https://www.defectdojo.org/)
- [CodeQL code-scanning from GitHub](https://docs.github.com/en/code-security/code-scanning)

### Dependencies upgrade/scan
- [renovatebot](https://docs.renovatebot.com/)
- [dependabot from GitHub](https://github.com/dependabot)

**Comment by mazipan** [comment]
Alternative guest:
- https://x.com/BukanYahya
- https://x.com/dwisiswant0
- https://x.com/saifulwebid

**Comment by rizafahmi** [comment]
Wah menarik, mari kita agendakan pembahasan security ini @ivankristianto @ekafyi

**Reply by mazipan** [reply]
Karena cukup banyak, mungkin bisa dipecah jadi beberapa sesi aja, takut mabok kalau dibahas di satu sesi.
Diskusi aja part2 mana yang prioritas mau dibahas lebih dulu, dan lainnya belakangan.
Lebih baik bahas 1 atau 2 hal tapi bisa sampai pesannya, daripada kebanyakan malah gak ada yang nyampe.

**Reply by rizafahmi** [reply]
Setuju Mas Irfan. Bisa jadi berseri ini. Mungkin bisa mulai dari yang paling dasar dulu aja, yg sifatnya low hanging fruit. Baru nanti level selanjutnya
